client/lib/user/utils.js

```diff
@@ -42,6 +44,12 @@ var userUtils = {
 	logout: function( redirect ) {
 		var logoutUrl = userUtils.getLogoutUrl( redirect );
 
+		beforeLogOutCallbacks.forEach( function( callback ) {
+			if ( 'function' === typeof callback ) {
+				callback();
```

Comment:
Should `try` / `catch` here to prevent errors in callbacks from affecting logout.